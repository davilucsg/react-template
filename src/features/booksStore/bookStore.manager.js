import BooksStoreService from './booksStore.service';
class BookStoreManager {
  async getBooks() {
    try {
      const respostaServico = await BooksStoreService.getBooks();
      return respostaServico.data;
    } catch (error) {
      console.log('error :', error);
    }
  }
}

export default new BookStoreManager();
