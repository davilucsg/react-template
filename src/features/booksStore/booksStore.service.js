import axios from 'axios';
const URL_API = 'https://workshopreact-a7f54.firebaseio.com/books.json/';
class BookStoreService {
  async getBooks() {
   return await axios.get(URL_API);
  }
}
export default new BookStoreService();
