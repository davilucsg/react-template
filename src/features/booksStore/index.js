export { default as BookStoreContainer } from './views/bookStore.container';
