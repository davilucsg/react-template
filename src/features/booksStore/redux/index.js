import * as bookStoreActions from './bookStore.actions';
import * as bookStoreOperations from './bookStore.operations';
export { default as bookStoreReducer } from './bookStore.reducers';
export { bookStoreActions, bookStoreOperations };
