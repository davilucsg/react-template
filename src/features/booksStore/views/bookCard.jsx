import React from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import FavoriteIcon from '@material-ui/icons/Favorite';
import { ShoppingCart } from '@material-ui/icons/';
import { Grid, Typography, IconButton, CardMedia } from '@material-ui/core';

import * as bookStoreActions from '../redux/bookStore.actions';

export class BookCard extends React.Component {
  incluiNoCarrinho = () => {
    this.props.incluiNoCarrinho(this.props.book);
  };

  incluiNalistaDesejos = () => {
    this.props.incluiNalistaDesejos(this.props.book);
  };

  render() {
    const { classes, book } = this.props;
    return (
      <Card className={classes.card}>
        <Grid container direction="column" justify="flex-end">
          <CardMedia
            id={`imgBook${book.id}`}
            onClick={() => alert(book.volumeInfo.title)}
            className={classes.cover}
            image={book.volumeInfo.imageLinks.smallThumbnail}
            title="Live from space album cover"
          />
          <CardContent className={classes.cardContent}>
            <Typography component="p" variant="subtitle1" id={`tituloLivro${book.id}`}>
              {book.volumeInfo.title}
            </Typography>
            <Typography variant="subtitle2" color="textSecondary" id={`autorLivro${book.id}`}>
              {book.volumeInfo.authors ? book.volumeInfo.authors[0] : ''}
            </Typography>
          </CardContent>
          <div>
            <IconButton
              aria-label="Add to favorites"
              id={`btnFavorite${book.id}`}
              onClick={this.incluiNalistaDesejos}
            >
              <FavoriteIcon />
            </IconButton>
            <IconButton
              aria-label="Share"
              id={`btnShopping${book.id}`}
              onClick={this.incluiNoCarrinho}
            >
              <ShoppingCart />
            </IconButton>
          </div>
        </Grid>
      </Card>
    );
  }
}

BookCard.propTypes = {
  classes: PropTypes.object.isRequired
};

const styles = theme => ({
  card: {
    height: 350,
    width: 250,
    cursor: 'pointer'
  },
  cardContent: {
    height: 75
  },
  cover: {
    height: 0.55 * 350,
    width: 250
  }
});

const mapDispatchToProps = {
  incluiNalistaDesejos: bookStoreActions.incluiNalistaDesejos,
  incluiNoCarrinho: bookStoreActions.incluiNoCarrinho
};

export default compose(
  withStyles(styles),
  connect(
    null,
    mapDispatchToProps
  )
)(BookCard);
