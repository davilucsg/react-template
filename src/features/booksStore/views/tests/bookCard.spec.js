import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { BookCard } from '../bookCard';
import { booksMock } from '../../mocks/booksStore.mock';
configure({ adapter: new Adapter() });

describe('Testa <BookCard />', function() {
  let incluiNoCarrinhoMock = jest.fn();
  let incluiNalistaDesejosMock = jest.fn();
  
  const bookMock = { ...booksMock.data[0] };
  let component;
  beforeEach(() => {
    component = shallow(
      <BookCard
        incluiNoCarrinho={incluiNoCarrinhoMock}
        incluiNalistaDesejos={incluiNalistaDesejosMock}
        classes={{}}
        book={bookMock}
      />
    );
  });

  it('Deve renderizar <BookCard /> ', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it(`Deve renderizar a imagem do book: imgBook${bookMock.id}`, () => {
    const imagemBook = component.find(`#imgBook${bookMock.id}`);
    expect(imagemBook.exists()).toEqual(true);
    expect(imagemBook.props().image).toEqual(bookMock.volumeInfo.imageLinks.smallThumbnail);
    // expect(imagemBook.props().image).toEqual('bla');
  });

  it(`Ao clicar na imagem do <BookCard /> deve chamar o window.alert() com a mensagem ${
    bookMock.volumeInfo.title
  }`, () => {
    global.alert = jest.fn();
    const imagemBook = component.find(`#imgBook${bookMock.id}`);
    imagemBook.simulate('click');
    expect(global.alert).toHaveBeenCalled();
    expect(global.alert).toBeCalledWith(bookMock.volumeInfo.title);
  });

  it(`Deve renderizar o botão de incluir em na lista e desejos: btnFavorite${bookMock.id}`, () => {
    expect(component.find(`#btnFavorite${bookMock.id}`).exists()).toEqual(true);
  });

  it(`Ao clicar no de icone de  incluir em na lista e desejos: btnFavorite${
    bookMock.id
  } a função props.incluiNalistaDesejos() deve ser executada `, () => {
    const botao = component.find(`#btnFavorite${bookMock.id}`);
    botao.simulate('click');
    expect(incluiNalistaDesejosMock).toHaveBeenCalled();
  });

  it(`Deve renderizar o botão de incluir no carrinho de compras: btnShopping${bookMock.id}`, () => {
    expect(component.find(`#btnShopping${bookMock.id}`).exists()).toEqual(true);
  });

  it(`Ao clicar no botão  de incluir no carrinho de compras: btnShopping${
    bookMock.id
  } a função props.incluiNoCarrinho() deve ser executada `, () => {
    const botao = component.find(`#btnShopping${bookMock.id}`);
    botao.simulate('click');
    expect(incluiNoCarrinhoMock).toHaveBeenCalled();
  });

  it(`Deve renderizar o titulo do livro ${bookMock.volumeInfo.title}`, () => {
    expect(component.find(`#tituloLivro${bookMock.id}`).exists()).toEqual(true);
    expect(
      component
        .find(`#tituloLivro${bookMock.id}`)
        .children()
        .text()
    ).toEqual(bookMock.volumeInfo.title);
  });

  it(`Deve renderizar o autor do livro ${bookMock.volumeInfo.authors[0]}`, () => {
    expect(component.find(`#autorLivro${bookMock.id}`).exists()).toEqual(true);
    expect(
      component
        .find(`#autorLivro${bookMock.id}`)
        .children()
        .text()
    ).toEqual(bookMock.volumeInfo.authors[0]);
  });
});
