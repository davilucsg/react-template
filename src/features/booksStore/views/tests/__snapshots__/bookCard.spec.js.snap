// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa <BookCard /> Deve renderizar <BookCard />  1`] = `
<WithStyles(Card)>
  <WithStyles(Grid)
    container={true}
    direction="column"
    justify="flex-end"
  >
    <WithStyles(CardMedia)
      id="imgBookH9lBCwAAQBAJ"
      image="http://books.google.com/books/content?id=H9lBCwAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
      onClick={[Function]}
      title="Live from space album cover"
    />
    <WithStyles(CardContent)>
      <WithStyles(Typography)
        component="p"
        id="tituloLivroH9lBCwAAQBAJ"
        variant="subtitle1"
      >
        Jane Eyre
      </WithStyles(Typography)>
      <WithStyles(Typography)
        color="textSecondary"
        id="autorLivroH9lBCwAAQBAJ"
        variant="subtitle2"
      >
        Charlotte BrontÃ«
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
    <div>
      <WithStyles(IconButton)
        aria-label="Add to favorites"
        id="btnFavoriteH9lBCwAAQBAJ"
        onClick={[Function]}
      >
        <pure(FavoriteIcon) />
      </WithStyles(IconButton)>
      <WithStyles(IconButton)
        aria-label="Share"
        id="btnShoppingH9lBCwAAQBAJ"
        onClick={[Function]}
      >
        <pure(ShoppingCartIcon) />
      </WithStyles(IconButton)>
    </div>
  </WithStyles(Grid)>
</WithStyles(Card)>
`;
